classic/helpers.go

```diff
@@ -0,0 +1,78 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package classic
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/ubuntu-core/snappy/dirs"
+)
+
+func mountpoint(path string) bool {
+	err := exec.Command("mountpoint", path).Run()
+	// man-page: zero if the directory is a mountpoint, non-zero if not
```

Comment:
I reworked this code to make it more robust. Its a bit tricky because mountpoint.c from util-linux is not a great interface. I think my version is pretty close to "as good as it gets". To improve further we would have to rewrite mountpoint.c in go (which may actually be a good idea).